File: @LV2/2.0/aboveaverage/aboveaverage.go
package main

import "fmt"

func avg(n int, score []float64) float64 {
	sum := 0.0
	for i := 0; i < n; i++ {
		sum += score[i]
	}
	return sum / float64(n)
}

func main() {
	var n, m int
	score := make([]float64, 1000)
	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &m)
		for j := 0; j < m; j++ {
			fmt.Scanf("%f", &score[j])
		}
		mean := avg(m, score)
		remain := 0.0

		for j := 0; j < m; j++ {
			if score[j] > mean {
				remain++
			}
		}
		fmt.Printf("%.3f%%\n", 100.0*remain/float64(m))
	}
}

File: @LV2/2.0/aboveaverage/aboveaverage.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut buff = String::new();

    stdin.read_line(&mut buff).unwrap();

    for line in stdin.lock().lines().map(|l| l.unwrap()) {
        let score: Vec<f32> = line
            .split_whitespace()
            .skip(1)
            .map(|n| n.parse().unwrap())
            .collect();

        let l = score.len() as f32;
        let mean: f32 = score.iter().sum::<f32>() / l;

        let rem = score.iter().filter(|&n| n > &mean).count() as f32;
        println!("{:.3}%", rem * 100.0 / l);
    }
}
